src/entities/Proposal/model.ts

```diff
@@ -235,120 +241,140 @@ export default class ProposalModel extends Model<ProposalAttributes> {
     const result = await this.query(SQL`
     SELECT COUNT(*) as "total"
     FROM ${table(ProposalModel)} p
-          ${conditional(!!filter.subscribed, SQL`INNER JOIN ${table(SubscriptionModel)} s ON s."proposal_id" = p."id"`)}
-          ${conditional(
-            !!filter.coauthor && !!filter.user,
-            SQL`INNER JOIN ${table(CoauthorModel)} c ON c."proposal_id" = p."id"`
-          )} 
+          ${conditional(!!subscribed, SQL`INNER JOIN ${table(SubscriptionModel)} s ON s."proposal_id" = p."id"`)}
+          ${conditional(!!coauthor && !!user, SQL`INNER JOIN ${table(CoauthorModel)} c ON c."proposal_id" = p."id"`)} 
           ${conditional(
-            !!filter.coauthor && !filter.user,
+            !!coauthor && !user,
             SQL`LEFT OUTER JOIN (
               select proposal_id, array_agg(address) coauthors
               from ${table(CoauthorModel)}
               where status = ${CoauthorStatus.APPROVED}
               group by proposal_id) c
           on p.id = c.proposal_id`
           )} 
-          ${conditional(
-            !!filter.search,
-            SQL`, ts_rank_cd(textsearch, to_tsquery(${tsquery(filter.search || '')})) AS "rank"`
-          )}
+          ${conditional(!!search, SQL`, ts_rank_cd(textsearch, to_tsquery(${tsquery(search || '')})) AS "rank"`)}
       WHERE "deleted" = FALSE 
       ${conditional(!!sqlSnapshotIds, SQL`AND p."snapshot_id" IN (${sqlSnapshotIdsJoin})`)} 
-      ${conditional(!!filter.user && !filter.coauthor, SQL`AND p."user" = ${filter.user}`)} 
+      ${conditional(!!user && !coauthor, SQL`AND p."user" = ${user}`)} 
+      ${conditional(
+        !!coauthor && !!user,
+        SQL`AND lower(c."address") = lower(${user}) AND (CASE WHEN p."finish_at" < NOW() THEN c."status" IN (${CoauthorStatus.APPROVED}, ${CoauthorStatus.REJECTED}) ELSE TRUE END)`
+      )}
+      ${conditional(!!type, SQL`AND "type" = ${type}`)} 
       ${conditional(
-        !!filter.coauthor && !!filter.user,
-        SQL`AND lower(c."address") = lower(${filter.user}) AND (CASE WHEN p."finish_at" < NOW() THEN c."status" IN (${CoauthorStatus.APPROVED}, ${CoauthorStatus.REJECTED}) ELSE TRUE END)`
+        !!subtype,
```

Comment:
It'd be good to extract this logic to a method:

```typescript
${conditional(!!subtype, SQL`AND (${this.getSubtypeQuery(subtype)})`)}
```

So you can use it in the other query of the Model as well.

```typescript
private static getSubtypeQuery(subtype: string) {
    return subtype === SubtypeAlternativeOptions.Legacy
      ? this.getLegacyGrantCategoryQuery()
      : SQL`p."configuration" @> '{"category": "${SQL.raw(subtype || '')}"}'`
  }
```

Also, since you are checking `!!subtype`, I think you don't need to
`SQL.raw(subtype || '')` and you can just `{SQL.raw(subtype)`

```typescript
private static getLegacyGrantCategoryQuery() {
    return join(
      Object.values(OldGrantCategory).map(
        (category) => SQL`p."configuration" @> '{"category": "${SQL.raw(category)}"}'`
      ),
      SQL` OR `
    )
  }
  ```

Reply:
It is true that I am checking `!!subtype` but that check is the p1 parameter of the `conditional(p1, p2)` function.  The function returns p2 if p1 is true, but whatever is used in p2 is evaluated regardless of whether p1 is true or not (it is not lazy like p1 && p2).

That said, the SQL.raw() function fails if the parameter passed to it is undefined.